channels_redis/core.py

```diff
@@ -132,13 +132,14 @@ def create_pool(self, index):
         if "address" in host:
             return aioredis.ConnectionPool.from_url(host["address"])
         elif "master_name" in host:
-            sentinels = host.pop("sentinels")
-            master_name = host.pop("master_name")
-            sentinel_kwargs = host.pop("sentinel_kwargs", None)
+            host_kwargs = deepcopy(host)
+            sentinels = host_kwargs.pop("sentinels")
+            master_name = host_kwargs.pop("master_name")
+            sentinel_kwargs = host_kwargs.pop("sentinel_kwargs", None)
```

Comment:
There is no need to have an other variable (`host_kwargs`) since from this point there will be no more use of the old `host` variable.

To guard the host main dict from being changed it should be enough to use `.copy` method (as it is done in the pubsub layer), thus the deepcopy is not required.

https://github.com/django/channels_redis/blob/a7094c58a15cbf6e621b2129253060fc80cfdfad/channels_redis/pubsub.py#L266

Also a comment line may be useful here.

```suggestion
            # copy host dict to prevent original dict from being modified
            host= host.copy()
            sentinels = host.pop("sentinels")
            master_name = host.pop("master_name")
            sentinel_kwargs = host.pop("sentinel_kwargs", None)
```